main.go

```diff
@@ -111,20 +111,48 @@ func (c *Controller) handleNewCSR(key string) error {
 		return nil
 	}
 
-	if pending := recentlyPendingCSRs(c.indexer); pending > maxPendingCSRs {
-		klog.Infof("ignoring all CSRs as too many recent pending CSRs seen: %d", pending)
-		return nil
+	machines, err := c.machines.List(metav1.ListOptions{})
```

Comment:
can we get all this to a function e.g `calculateMaxAllowedPendingCSRs()` and unit test it?